src/osal/osal.dox

```diff
@@ -237,27 +245,22 @@
  * OS / Platform | Thread unblocked upon post/increment
  * ------------- | ------------------------------------
  * chibios_arm   | FIFO if `CH_CFG_USE_SEMAPHORES_PRIORITY = false` (default), otherwise thread with highest priority
+ * epos_arm      | FIFO
  * linux_arm     | ?
  * linux_x64     | ?
- * linux_arm_tfc | Thread with highest priority
- * linux_x64_tfc | Thread with highest priority
+ * linux_arm_tfc | Random
+ * linux_x64_tfc | Random
```

Comment:
TFC creates the illusion of a machine with infinite processing power and an infinite number of CPU cores. If processing power tends to infinity, then time consumed by machine instructions tends to zero.

TFC increments the system time only if all threads are blocked. That means that all threads will complete their work in zero time regardless of their priority. Thread priority is therefore meaningless in a unittest environment using TFC.

Software should be designed in a way that it behaves correctly and provides the intended functionality regardless of thread priorities, or at least that it tries to do so. However, in a productive environment, if scheduling policies, thread priorities, and sometimes even CPU affinity are not set properly, then the software may not fulfil timing or performance requirements (what may be considered as failure), or the software may use the available CPU bandwidth inefficiently (e.g. lots of context switches).

Since thread priority is meaningless in a unittest environment using TFC, the order in which threads are unblocked from a convar, semaphore or mutex is also not relevant.

TFC also states that if multiple threads are ready (runnable) at the same time, then the order in which they are scheduled is undefined (see GPCC -> OSAL -> Time Flow Control -> chapter "Reproducibility of Unit tests"). Any attempt to model the scheduling of threads in TFC makes no sense, because each OS has its own flavors and quirks if you zoom deep enough into timing behavior. Not to forget about the impact of other processes...

Maybe the documentation use "not relevant" instead of "random"?

Reply:
OK, I will render "random" more precisely